### PR TITLE
feat(cli): Add flags for configuring external IPs

### PR DESCRIPTION
**Pull request by lwander**
Also change format of consul & vault flags

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -339,10 +340,27 @@ hal config security [parameters] [subcommands]
...
+
+Edit top-level spinnaker security settings.
+
+#### Usage
+```
+hal config security edit [parameters]
+```
+#### Parameters
+ * `--api-address`: The IP address the API is to be served from. If this is anything other than localhost it must be provisioned and owned by the cloud provider Spinnaker is being deployed to. The deployment of Spinnaker will automatically provision a load balancer that binds to this address.
+ * `--api-domain`: A domain that resolves to the address supplied to --api-address. This is reqapired for generating valid SSL keypairs based on whatever CA you provide Spinnaker.
```
Comment (duftler) [inline docs/commands.md]: s/reqapired/required

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/EditSecurityCommand.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  @Parameter(
+      names = "--ui-domain",
+      description = "A domain that resolves to the address supplied to --ui-address. This is required for generating "
+          + "valid SSL keypairs based on whatever CA you provide Spinnaker."
+  )
+  private String uiDomain;
+
+  @Parameter(
+      names = "--api-domain",
+      description = "A domain that resolves to the address supplied to --api-address. This is reqapired for generating "
```
Comment (duftler) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/EditSecurityCommand.java]: s/reqapired/required

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/EditSecurityCommand.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    security.setUiAddress(apiDomain);
+
+    if (originalHash == security.hashCode()) {
+      AnsiUi.failure("No changes supplied.");
+      return;
+    }
+
+    new OperationHandler<Void>()
+        .setOperation(Daemon.setSecurity(currentDeployment, !noValidate, security))
+        .setFailureMesssage("Failed to edit security.")
```
Comment (duftler) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/EditSecurityCommand.java]: You're calling `.setFailureMessage()` twice here.

File: docs/commands.md (hunk 1 above)
Comment (lwander) [inline docs/commands.md]: Oh boy I should get some sleep

**Pull request merged**